core/src/validator.rs

```diff
@@ -162,6 +162,7 @@ pub struct ValidatorConfig {
     pub warp_slot: Option<Slot>,
     pub accounts_db_test_hash_calculation: bool,
     pub accounts_db_skip_shrink: bool,
+    pub no_serve_snapshots: bool,
```

Comment:
Can this be `serve_snapshots` instead, and default to `true`? I'd like to avoid double negatives where possible.

Reply:
"Is the only way to not serve snapshots to skip pushing the snapshot hashes to gossip?"

That was the way I figured out should work..  This is my first Solana code, and I don't have a deep enough grasp of the codebase to have found another way. But if you aren't serving snapshots (because your hashes won't be what's expected by other validators), it seems like a mistake to publish anything about your snapshot hash.

Reply:
The hashes WILL be the same. But, the contents will be different and if and if an older version receiver were to recalculate the hash from the snapshot, the old version receiver would calculate the wrong hash and be unhappy with the snapshot.